### PR TITLE
feat: adds aws-sdk shapevisitors

### PR DESCRIPTION
**Pull request by ShubhamChaturvedi7**
*Issue #, if available:*

*Description of changes:*
Adds aws-sdk shapevisitors
Addresses the timestamp comment from here: https://github.com/smithy-lang/smithy-dafny/pull/672

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/AwsSdkToDafnyShapeVisitor.java
Comment (lucasmcdonald3) [inline codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/AwsSdkToDafnyShapeVisitor.java]: Hm. I'm not sure how I feel about this. 
This is required for DDB to work, right?
Is there a way to maybe return a runtime Go error instead of some "success but not really" value?

File: codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/DafnyToAwsSdkShapeVisitor.java
Comment (lucasmcdonald3) [inline codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/DafnyToAwsSdkShapeVisitor.java]: Error?

File: codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/ShapeVisitorHelper.java
Comment (lucasmcdonald3) [inline codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/ShapeVisitorHelper.java]: Wait. Is this branch ever true?
An AWS SDK shape with a ReferenceTrait?
To that effect: can `isConfigShape` be true for AWS SDK shapes?

**Review by lucasmcdonald3 — COMMENTED** [review]
Looks good -- I also looked at generated code over [here](https://github.com/aws/aws-cryptographic-material-providers-library/tree/Golang/dev) and it's fine.

I'm most interested in the TimestampShape. Could you return some "error: not supported" value at runtime instead of "success: no value"?

File: codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/DafnyToAwsSdkShapeVisitor.java
Comment (lucasmcdonald3) [reply codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/DafnyToAwsSdkShapeVisitor.java]: Chatted offline -- I am personally fine shipping without this for any beta releases, but we should address this before a full release

File: codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/AwsSdkToDafnyShapeVisitor.java
Comment (lucasmcdonald3) [reply codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/AwsSdkToDafnyShapeVisitor.java]: Chatted offline -- I am personally fine shipping without this for any beta releases, but we should address this before a full release

File: codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/ShapeVisitorHelper.java
Comment (ShubhamChaturvedi7) [reply codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/ShapeVisitorHelper.java]: Nope, removed both of them.

**Comment by ShubhamChaturvedi7** [comment]
> Looks good -- I also looked at generated code over [here](https://github.com/aws/aws-cryptographic-material-providers-library/tree/Golang/dev) and it's fine.
> 
> I'm most interested in the TimestampShape. Could you return some "error: not supported" value at runtime instead of "success: no value"?

Implemented the full thing instead of erroring out since it is required in KeyStore.

File: codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/localservice/nameresolver/DafnyNameResolver.java
Comment (lucasmcdonald3) [inline codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/localservice/nameresolver/DafnyNameResolver.java:228]: Is there any way to just reuse this implementation? https://github.com/smithy-lang/smithy-dafny/blob/180ad67ad9f80a8b5f9272b06dbbdcce24d4f369/codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/utils/DafnyNameResolverHelpers.java#L46

File: codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/localservice/nameresolver/SmithyNameResolver.java
Comment (lucasmcdonald3) [inline codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/localservice/nameresolver/SmithyNameResolver.java:32]: Me neither.

**Review by lucasmcdonald3 — APPROVED** [review]

File: codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/localservice/nameresolver/DafnyNameResolver.java
Comment (ShubhamChaturvedi7) [reply codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/localservice/nameresolver/DafnyNameResolver.java:228]: Let me do this once the code base has stabilized, during the final review before main merge.

File: codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/localservice/nameresolver/SmithyNameResolver.java
Comment (ShubhamChaturvedi7) [reply codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/localservice/nameresolver/SmithyNameResolver.java:32]: It's also likely that this has been lifted directly from the Python codegen :)

**Pull request merged**
